packages/vite/src/node/optimizer/index.ts

```diff
@@ -1174,25 +1168,14 @@ export async function extractExportsData(
     const transformed = await transformWithEsbuild(entryContent, filePath, {
       loader,
     })
-    // Ensure that optimization won't fail by defaulting '.js' to the JSX parser.
-    // This is useful for packages such as Gatsby.
-    esbuildOptions.loader = {
-      '.js': 'jsx',
-      ...esbuildOptions.loader,
-    }
```

Comment:
I'm not sure if it's intentional, but this code mutates the global `esbuildOptions` config when using this function and hitting this code path. I've removed it as it should already be handled at https://github.com/vitejs/vite/blob/094880306e3474c2a03d8b711525a091a128fb31/packages/vite/src/node/optimizer/index.ts#L729-L736